Providers/Scripts/2.4x-2.5x/Scripts/nxAvailableUpdates.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+# ===================================
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See license.txt for license information.
+# ===================================
+
+import subprocess
+import sys
+import imp
+import re
+import copy
+import fnmatch
+
+protocol = imp.load_source('protocol', '../protocol.py')
+nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+helperlib = imp.load_source('helperlib', '../helperlib.py')
+LG = nxDSCLog.DSCLog
+
+# [ClassVersion("1.0.0"),FriendlyName("nxAvailableUpdates"),SupportsInventory()]
+# class MSFT_nxAvailableUpdatesResource : OMI_BaseResource
+# {
+#   [Key, InventoryFilter] string Name;
+#   [read] string BuildDate;
+#   [read] string Repository;
+#   [read] string Version;
+#   [read] string Architecture;
+ 
+# };
+
+def Inventory_Marshall(Name):
+    if Name is not None and len(Name) > 0:
+        Name = Name.encode('ascii','ignore')
+    retval, pkgs = GetUpdates(Name)
+    for p in pkgs:
+        p['Name'] = protocol.MI_String(p['Name'])
+        p['BuildDate'] = protocol.MI_String('BuildDate')
+        p['Repository'] = protocol.MI_String(p['Repository'])
+        p['Version'] = protocol.MI_String(p['Version'])
+        p['Architecture'] = protocol.MI_String(p['Architecture'])
+    Inventory=protocol.MI_InstanceA(pkgs)
+    retd = {}
+    retd["__Inventory"] = Inventory
+    return retval, retd
+
+def GetUpdates(Name):
+    d={}
+    mgr = GetPackageManager()
+    if mgr == None:
+        Print("ERROR: Unable to find one of 'apt', 'yum', or 'zypper'.")
+        LG().Log('ERROR', "Unable to find one of 'apt', 'yum', or 'zypper'.")
+        return -1, d
+    if mgr == 'apt':
+        d = GetAptUpdates(Name)
+    elif mgr == 'yum':
+        d = GetYumUpdates(Name)
+    elif mgr == 'zypper':
+        d = GetZypperUpdates(Name)
+    return 0, d
+
+def GetAptUpdates(Name):
+    # Format:
+    # Inst python3.5 (3.5.0-3 Ubuntu:15.10/wily [amd64])
+    # Inst sosreport [3.2-2ubuntu1] (3.2-2ubuntu1.1 Ubuntu:15.10/wily-updates [amd64])
+
+    updates_list = []
+    d={}
+    cmd = 'LANG=en_US.UTF8 apt-get -s dist-upgrade | grep "^Inst"'
+    code, out = RunGetOutput(cmd, False, False)
+    srch_txt=r'Inst[ ](.*?)[ ].*?[(](.*?)[ ](.*?)[ ]\[(.*?)\]'
+    srch=re.compile(srch_txt, re.M|re.S)
```

Comment:
Pretty sure you don't want both `re.M` and `re.S` here since each result is expected to be on one line.